js-miniapp-bridge/src/common-bridge.ts

```diff
@@ -482,6 +482,17 @@ export class MiniAppBridge {
       );
     });
   }
+
+  setSecondaryWebviewClose(): Promise<string> {
+    return new Promise<string>((resolve, reject) => {
```

Comment:
I don't think a Promise is the right way to handle this. A promise is used for a one-time asynchronous operation which can either succeed or fail. However "closing the secondary webview" is an event (not an operation) that could occur multiple times in the lifetime of a mini app and doesn't have a failure state.

So instead, I think we should either:

1. Let them pass a function which we save and then call when the JS SDK receives the "close" event. i.e. `setWebviewClosedListener( (message) => { console.log("webview closed") })`. This would probably work similar to how we have setup the `exec` function - so we might need to create a separate bridge function on the native side for passing this event.
2. OR Create a [synthetic event](https://developer.mozilla.org/en-US/docs/Web/Events/Creating_and_triggering_events) that the Mini App can listen for such as `window.addEventListener("miniAppWebviewClosed", () => {} )`.

I think option 2 would require us to also implement an interface similar to optional 1 anyways, so maybe just doing 1 is okay.

What do you think?

Reply:
@corycaywood i will look into it, thanks for the suggestions